providers/azure.go

```diff
@@ -53,6 +53,14 @@ var (
 		Path:   "/v1.0/me",
 	}
 
+	// Default Groipe URL for Azure.
+	// Pre-parsed URL of https://graph.microsoft.com/v1.0/me.
+	azureDefaultGroupURL = &url.URL{
+		Scheme: "https",
+		Host:   "graph.microsoft.com",
+		Path:   "/v1.0/me/transitiveMemberOf/microsoft.graph.group",
```

Comment:
This path requires high privileges like `Directory.Read.All` otherwise it won't work(e.g. will retrieve `nil` as value for `displayName` key).

One approach that might not need extended permissions could be to use `/v1.0/me/getMemberGroups` instead, but in this case there are no trivial ways to find if a group is part of another group. Authentication might fail in this case when it shouldn't(e.g. a user will be denied even if he is part of a group that is part of the group configured in the oauth2-proxy `allowed-group` list).

Reply:
Even `/v1.0/me/getMemberGroups` requires `Directory.Read.All` permissions based on my testing 😞

Reply:
Why not using `id` instead of `displayName`(in `getGroupsFromJSON` method) when checking if a user is part of group or not. In this way we don't need elevated permissions anymore.